tools/base/fpgad/config_file.c

```diff
@@ -183,7 +187,299 @@ STATIC cfg_plugin_configuration *alloc_configuration(bool enabled,
 
 	return p;
 }
-#endif
+
+STATIC cfg_vendor_device_id *
+cfg_process_plugin_devices(const char *name,
+			   json_object *j_devices)
+{
+	int i;
+	int devs;
+	cfg_vendor_device_id *head = NULL;
+	cfg_vendor_device_id *id = NULL;
+	uint16_t vendor_id;
+	uint16_t device_id;
+	const char *s;
+	char *endptr;
+
+	if (!json_object_is_type(j_devices, json_type_array)) {
+		LOG("'devices' JSON object not array.\n");
+		return NULL;
+	}
+
+	devs = json_object_array_length(j_devices);
+	for (i = 0 ; i < devs ; ++i) {
+		json_object *j_dev = json_object_array_get_idx(j_devices, i);
+		json_object *j_vid;
+		json_object *j_did;
+
+		if (!json_object_is_type(j_dev, json_type_array)) {
+			LOG("%s 'devices' entry %d not array.\n",
+				name, i);
+			goto out_free;
+		}
+
+		if (json_object_array_length(j_dev) != 2) {
+			LOG("%s 'devices' entry %d not array[2].\n",
+				name, i);
+			goto out_free;
+		}
+
+		j_vid = json_object_array_get_idx(j_dev, 0);
```

Comment:
You think this (getting or parsing an int from a jarray at index I) could be refactored into a function?

Reply:
Sure.. but for 2 items, I'm not sure it's worth the trouble.